### DISCUSSION TITLE
弱弱的问一下，hp会更新rule-set

### DISCUSSION THREAD

**Discussion by elric8** [immortalwrt/homeproxy#74] [Q&A]
两个问题
1，因为家里的路由器，怕更新了，sing-box1.8+ geosite移除了，还要重新配置
2，hp的配置文件支持手搓不，像配置rule-set，不能手搓有点麻烦
感谢大佬

**Comment by muink** [comment]
SB 有缺省的GEO二进制规则, 可以直接用: [sing-geoip](https://github.com/SagerNet/sing-geoip/tree/rule-set), [sing-geosite](https://github.com/SagerNet/sing-geosite/tree/rule-set)
同样也可以手搓配置: [source-format](https://sing-box.sagernet.org/configuration/rule-set/source-format/), [headless-rule](https://sing-box.sagernet.org/configuration/rule-set/headless-rule/)
另外手搓的配置也支持使用 `sing-box rule-set compile` 命令转换为体积更小的二进制文件

**Reply by rianjskis** [reply]
您好我在自定义路由模式中的规则集如此填写可以直接使用吗
![2](https://github.com/immortalwrt/homeproxy/assets/86937445/3935d08d-0570-4649-b68d-fb472a1103bc)
如果不能直接使用需要用到geosite.db,geoip.db的话，请教一下在路由规则以及DNS规则中使用geoip-cn geosite-cn该在那个选项填上，
![3](https://github.com/immortalwrt/homeproxy/assets/86937445/21d253f0-1b8f-4ca0-bca2-5895873c32ac)
![4](https://github.com/immortalwrt/homeproxy/assets/86937445/56aa9245-8ef1-4c0d-86e7-f859536bc1d6)
![5](https://github.com/immortalwrt/homeproxy/assets/86937445/c2e2d401-24ce-4d69-9cde-1f5696a0e05d)

**Reply by muink** [reply]
匹配规则集那里 选择已有的规则集

**Reply by rianjskis** [reply]
> 匹配规则集那里 选择已有的规则集

您的意思是可以直接使用.srs对吗，但是我这样选择路由规则以及dns规则会出现没网，/etc/homeproxy/ruleset/ 里面也是空的,测试百度是通的，谷歌是不通的。

**Reply by rianjskis** [reply]
![2](https://github.com/immortalwrt/homeproxy/assets/86937445/309af666-26bd-43f2-ab0a-4bb11f3a5924)
![3](https://github.com/immortalwrt/homeproxy/assets/86937445/f5050491-d345-4e3a-b251-860c1950ae00)
![4](https://github.com/immortalwrt/homeproxy/assets/86937445/22ea8e80-8c43-48d8-b22d-7461a20471fa)
![5](https://github.com/immortalwrt/homeproxy/assets/86937445/49080e0c-b0a2-4308-a187-6eefa8d83f62)
![6](https://github.com/immortalwrt/homeproxy/assets/86937445/c1b31bbe-e5f0-4461-b46b-c45b1963ee6d)
![7](https://github.com/immortalwrt/homeproxy/assets/86937445/593390b5-8305-4901-83b7-47070476c80b)
这是我的选项配置，访问控制为默认的，我在DNS规则中设置geosite-geolocation-!cn走谷歌DNS，geosite-cn走阿里DNS解析，路由规则geosite-cn,geoip-cn,私有IP地址走直连，geosite-geolocation-!cn走代理，默认兜底走代理。

**Reply by muink** [reply]
`/etc/homeproxy/ruleset/`是留给用户放本地规则的, 空的是正常的
请排查url是否正确 `https://raw.githubusercontent.com/SagerNet/sing-geosite/rule-set/geosite-category-ads.srs`
日志是否有异常消息

**Reply by rianjskis** [reply]
规则集链接没问题我电脑与手机都是用的这个配置，目前来看.已经能启动,也没有报错日志，只是有些网站会特别的慢.请教一下在其他端这样实现的any DNS规则在homeproxy该如何写呢，另外可以使用fakeip吗
      {
        "outbound": "any",
        "server": "dns_resolver"
      },

**Reply by muink** [reply]
dns规则页面 `Match outbound` --> `Any`, 权重排到 gn 上方
hp不支持fakeip功能

**Reply by rianjskis** [reply]
> dns规则页面 `Match outbound` --> `Any`, 权重排到 gn 上方
> hp不支持fakeip功能

好的感谢解答

**Reply by onlyfirer** [reply]
dns 规则页面Match outbound--> Any这个什么意思？没有看明白

**Reply by muink** [reply]
匹配任何对 `outbound entity` 的 `server` 的域名的 DNS 解析请求
比如要连接下面这个节点, 就需要把 `host.need.resolve.com` 通过 DNS server 解析为 IP
``` json
{
  "type": "shadowsocks",
  "tag": "ss-out",
  "server": "host.need.resolve.com",
  "server_port": 1080,
  "method": "aes-128-gcm",
  "password": "8JCsPssfgS8tiRwiMlhARg"
}
```

它通常的作用是设置一个最高优先(除 block 外, block 优先级可以更高)规则, 用来将节点的域名解析为 IP

**Reply by onlyfirer** [reply]
如何配置才能完全防dns泄漏？

**Reply by muink** [reply]
sing-box默认的配置就已经防dns泄漏了

**Reply by onlyfirer** [reply]
homeproxy这个配置吗？
